extensions/warp-ipfs/src/thumbnail.rs

```diff
@@ -350,3 +324,46 @@ impl ThumbnailGenerator {
         task.await.map_err(anyhow::Error::from)?
     }
 }
+
+pub fn generate_thumbnail<R: BufRead + Seek>(
```

Comment:
You should be able to use `Read` instead of `BufRead`.

Reply:
imagereader requires a bufread